jupyter/d2l-java/chapter_convolutional-modern/alexnet.ipynb

```diff
@@ -0,0 +1,730 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "1"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "NDManager manager = NDManager.newBaseManager();\n",
+    "\n",
+    "SequentialBlock block = new SequentialBlock();\n",
+    "// Here, we use a larger 11 x 11 window to capture objects. At the same time,\n",
+    "// we use a stride of 4 to greatly reduce the height and width of the output.\n",
+    "//Here, the number of output channels is much larger than that in LeNet\n",
+    "block\n",
+    "                .add(Conv2D.builder()\n",
+    "                        .setKernelSize(new Shape(11, 11))\n",
+    "                        .optStride(new Shape(4, 4))\n",
+    "                        .setFilters(96).build())\n",
+    "                .add(Activation::relu)\n",
+    "                .add(Pool.maxPool2DBlock(new Shape(3, 3), new Shape(2, 2)))\n",
+    "        // Make the convolution window smaller, set padding to 2 for consistent\n",
+    "        // height and width across the input and output, and increase the\n",
+    "        // number of output channels\n",
+    "                .add(Conv2D.builder()\n",
+    "                        .setKernelSize(new Shape(5, 5))\n",
+    "                        .optPadding(new Shape(2, 2))\n",
+    "                        .setFilters(256).build())\n",
+    "                .add(Activation::relu)\n",
+    "                .add(Pool.maxPool2DBlock(new Shape(3, 3), new Shape(2, 2)))\n",
+    "        // Use three successive convolutional layers and a smaller convolution\n",
+    "        // window. Except for the final convolutional layer, the number of\n",
+    "        // output channels is further increased. Pooling layers are not used to\n",
+    "        // reduce the height and width of input after the first two\n",
+    "        // convolutional layers\n",
+    "                .add(Conv2D.builder()\n",
+    "                        .setKernelSize(new Shape(3, 3))\n",
+    "                        .optPadding(new Shape(1, 1))\n",
+    "                        .setFilters(384).build())\n",
+    "                .add(Activation::relu)\n",
+    "                .add(Conv2D.builder()\n",
+    "                        .setKernelSize(new Shape(3, 3))\n",
+    "                        .optPadding(new Shape(1, 1))\n",
+    "                        .setFilters(384).build())\n",
+    "                .add(Activation::relu)\n",
+    "                .add(Conv2D.builder()\n",
+    "                        .setKernelSize(new Shape(3, 3))\n",
+    "                        .optPadding(new Shape(1, 1))\n",
+    "                        .setFilters(256).build())\n",
+    "                .add(Activation::relu)\n",
+    "                .add(Pool.maxPool2DBlock(new Shape(3, 3), new Shape(2, 2)))\n",
+    "                .add(Blocks.batchFlattenBlock())\n",
```

Comment:
Why flatten in here?

Reply:
So in LeNet, which had a similar implementation, the network flattens a 4-D image to a 2-D image when going from a Conv2D to a Dense layer. I assumed similar thing is happening here too when we see/print the shape changes in the d2l book.